opm/autodiff/BlackoilWellModel_impl.hpp

```diff
@@ -37,12 +37,29 @@ namespace Opm {
     template<typename TypeTag>
     void
     BlackoilWellModel<TypeTag>::
-    beginReportStep(const int timeStepIdx)
+    beginReportStep(const int timeStepIdx,const double simulationTime, bool enableWellTests)
     {
         const Grid& grid = ebosSimulator_.vanguard().grid();
         const auto& defunct_well_names = ebosSimulator_.vanguard().defunctWellNames();
         const auto& eclState = ebosSimulator_.vanguard().eclState();
         wells_ecl_ = schedule().getWells(timeStepIdx);
+        const auto& wtest_config = schedule().wtestConfig(timeStepIdx);
+
+        // Check if the well is scheduled for a test
+        if (wtest_config.size() > 0 && enableWellTests) {
+            for (auto& well : wtest_config.getWells()) {
+                wellTestState_.addClosedWell(well.name, WellTestConfig::Reason::ECONOMIC, 0);
+            }
+            const auto& wellsForTesting = wellTestState_.update(wtest_config, simulationTime);
+            if (wellsForTesting.empty()) {
+                enableWellTests = false;
+            }
+            for (auto& testWell : wellsForTesting) {
```

Comment:
This looks incorrect to me? The `wellsForTesting` container contains a list of wells *which should be tested for a possible reopen*. Here it seems like the wells which *should be tested* - are just reopened unconditionally? One of the purposes of the `WellTestState` class was to facilitate that; so the code could be:
```C++
for (const auto& pair : wellsForTesting) {
     const std::string& well_name = pair.first;
     auto reason = pair.second;

     // Some logic to check if the well can be opened again; 
     // I do not really know how that should be implemented.
     if (determined_reason_no_longer_valid) 
        wellTestState.drop( well_name, reason);
}
```